src/services/swap_service.ts

```diff
@@ -471,7 +471,7 @@ export class SwapService {
                 },
             ]);
         }
-        if (apiKey !== undefined && (isETHSell || from !== undefined)) {
+        if (apiKey !== undefined && (isETHSell || from !== undefined || (rfqt && rfqt.isIndicative))) {
```

Comment:
Can you explain this flow to me? why is this change needed?

Reply:
This conditional says, "if there's an API key AND (if either it's a forwarder tx (`isETHSell`) OR there's a taker address (`from`) OR it's an indicative quote".

Without this addition, an indicative quote request without a taker address cannot flow through here, so `_rfqt` remains undefined, so RFQT never gets enabled downstream.

Reply:
This now makes sense for me, however, I'm a little concerned that whoever next touches this code won't understand the logic. Is it worth adding some comments explaining the reasoning behind L479? alternatively, we could refactor the `if` statement to use self-explaining variablenames. Example:

```
if (hasAPIKey && (isETHSell || hasTakerAddress || isIndicativeQuote)) {

}
```